plugins/out_cloudwatch_logs/cloudwatch_logs.c

```diff
@@ -333,7 +334,12 @@ static int cb_cloudwatch_init(struct flb_output_instance *ins,
     ctx->cw_client->proxy = NULL;
     ctx->cw_client->static_headers = &content_type_header;
     ctx->cw_client->static_headers_len = 1;
-    ctx->cw_client->extra_user_agent = (char *) ctx->extra_user_agent;
+    tmp_sds = flb_sds_create(ctx->extra_user_agent);
```

Comment:
this simplifies things to let cloudwatch use separate memory for user-agent

Reply:
oops/thanks

Reply:
I think this is a good idea! Just was requesting change on deleting the unused variable free_user_agent

Reply:
fixed

Reply:
I meant to reply to the other comment

Reply:
which I fixed